File: src/main/java/itmo/is/project/model/Resource.java
package itmo.is.project.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

@Entity
@Table(name = "resource")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Resource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @NotBlank
    @Size(max = 32)
    @Column(name = "name", length = 32, nullable = false, unique = true, updatable = false)
    private String name;
}

File: src/main/kotlin/itmo/isproject/model/resource/Resource.kt
package itmo.isproject.model.resource

import itmo.isproject.model.IntIdEntity
import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

@Entity
@Table(name = "resource")
class Resource(

    @NotBlank
    @Size(max = 32)
    @Column(name = "name", length = 32, nullable = false, unique = true, updatable = false)
    var name: String? = null

) : IntIdEntity()
